lib/cosmos/interfaces/interface.rb

```diff
@@ -372,7 +372,7 @@ def convert_data_to_packet(data)
     # @param packet [Packet] Packet to extract data from
     # @return data
     def convert_packet_to_data(packet)
-      packet.buffer(false)
+      packet.buffer(true) # Copy buffer so logged command isn't modified
```

Comment:
Can you expand on this? Commands are logged before sending to the interface. Then the write routine allows all the write protocols to write_packet which could potentially modify the packet. Then we convert_packet_to_data and then allow all the write protocols to write_data which could potentially modify the data. What are we protecting against here?